api/app/controllers/review.controller.ts

```diff
@@ -222,3 +226,76 @@ export async function submit(req: Request, res: Response) {
     });
   }
 }
+
+
+export const getAllReviewsForAdmin = async (req: Request, res: Response) => {
+  const id = Number(req.query.id);
+  const submitted_at = req.query.submitted_at as string | undefined;
+  const transcript_id = req.query.transcript_id as string | undefined;
+  const user_id = req.query.user_id as string | undefined;
+  const merged_at = req.query.merged_at as string | undefined;
+  const page: number = Number(req.query.page) || DB_START_PAGE;
+  const limit: number = Number(req.query.limit) || DB_QUERY_LIMIT;
+
+
+  const condition: {
+    [key: string | number]: any;
+  } = {};
+
+  // Add conditions if query parameters exist
+  if (Boolean(id)) {
+    condition.id = { [Op.eq]: id };
+  }
+  if (Boolean(submitted_at)) {
+    condition.submittedAt = {
+      [Op.gte]: submitted_at ? new Date(submitted_at) : null,
+      [Op.lte]: submitted_at ? new Date(submitted_at) : null,
+    };
+  }
+  if (Boolean(transcript_id)) {
+    condition.transcriptId = { [Op.eq]: transcript_id };
+  }
+  if (Boolean(user_id)) {
+    condition.userId = { [Op.eq]: user_id };
+  }
+  if (Boolean(merged_at)) {
+    condition.mergedAt = {
+      [Op.gte]: merged_at ? new Date(merged_at) : null,
+      [Op.lte]: merged_at ? new Date(merged_at) : null,
```

Comment:
why does this condition occur twice? Also we don't need the ternary check for `merged_at` since we already did a check on line 261

Reply:
There are two different conditions, one is checking all the reviews where the date specified is less than the day we are checking for the records and the other is greater than it.  So if the pass is on 12-10-2023 on 16-11-2023(today's date) I should get all the records within those days.  Except we want to just filter based on specific days. 

IMO it is harmless as is but the ternary can be updated.